nengo_loihi/compartments.py

```diff
@@ -105,45 +103,18 @@ def __init__(self, n, label=None, location='core'):
         self.noiseExp0 = 0
         self.noiseAtDendOrVm = 0
 
-        self.synapses = []
-        self.named_synapses = {}
-        self.axons = []
-        self.named_axons = {}
-        self.probes = []
-
-        assert location in ('core', 'cpu')
-        self.location = location
-
     def __str__(self):
         return "%s(%s)" % (
             type(self).__name__, self.label if self.label else '')
 
     def add_synapses(self, synapses, name=None):
-        """Add a Synapses object to ensemble."""
-
-        assert synapses.group is None
-        synapses.group = self
-        self.synapses.append(synapses)
-        if name is not None:
-            assert name not in self.named_synapses
-            self.named_synapses[name] = synapses
+        self.synapses.add(synapses, name=name)
 
     def add_axons(self, axons, name=None):
-        """Add a Axons object to ensemble."""
-
-        assert axons.group is None
-        axons.group = self
-        self.axons.append(axons)
-        if name is not None:
-            assert name not in self.named_axons
-            self.named_axons[name] = axons
+        self.axons.add(axons, name=name)
 
     def add_probe(self, probe):
-        """Add a Probe object to ensemble."""
-        if probe.target is None:
-            probe.target = self
-        assert probe.target is self
-        self.probes.append(probe)
+        self.probes.add(probe)
```

Comment:
add_synapses/add_axons/add_probe seem a bit redundant now, could we just get rid of them and change the corresponding calls `model.add_axons(...)` -> `model.axons.add(...)`?

Reply:
So it turns out we were relying on this. So I've reverted the change, and put the (broken) changed branch at `refactor-files-addfunctions`.